meson.build

```diff
@@ -140,14 +140,14 @@ subdir('inc')
 subdir('src')
 subdir('tools')
 
-if get_option('examples')
```

Comment:
test/example both should be at the same level as the apps. How the examples are successfully buildable...

Reply:
> test/example both should be at the same level as the apps. How the examples are successfully buildable...

I am not sure, if I understand what do you mean _at the same level as the apps_

Reply:
@SergeyLebedkin The question is why the build order of 'example/test' needs to be switched; they shouldn't affect each other.

Reply:
@hermet 
Its just a theory, but I think that problem is in _compiler_flags_ variable, that is the same for examples and test build script.
If examples builds before test, its meson.build script add **['-x', 'objective-c++']** flags, but for test they should not be there

Reply:
@hermet 
please, review the new variant